### PR TITLE
quic: add quic stream reset error code stats.

### PR DESCRIPTION
**Pull request by RenjieTang**
Commit Message: implement lazily initialized counters to log quic stream reset error code stats.
Additional Description:
Risk Level: Low
Testing: unit tests
Docs Changes: n/a
Release Notes: n/a.

### CODE REVIEW

File: source/common/quic/quic_stat_names.cc [hunk 1]
```diff
@@ -38,6 +42,20 @@ void QuicStatNames::chargeQuicConnectionCloseStats(Stats::Scope& scope,
...
+                                                    bool from_self, bool is_upstream) {
+  ASSERT(&symbol_table_ == &scope.symbolTable());
+
+  if (error_code > quic::QUIC_STREAM_LAST_ERROR) {
+    error_code = quic::QUIC_STREAM_LAST_ERROR;
+  }
+
+  const Stats::StatName stream_error = resetStreamErrorStatName(error_code);
+  incCounter(scope, {http3_prefix_, is_upstream ? upstream_ : downstream_,
+                     from_self ? from_self_ : from_peer_, stream_error});
```
Comment (RyanTheOptimist) [inline source/common/quic/quic_stat_names.cc:56]: nit: Up to you, but I think I would put ()s around the two "?:" expressions:

  incCounter(scope, {http3_prefix_, (is_upstream ? upstream_ : downstream_),
                     (from_self ? from_self_ : from_peer_), stream_error});

**Review by RyanTheOptimist — APPROVED** [review]
Nice! Looks good.

**Review by alyssawilk — COMMENTED** [review]
Code looks great!  

Let's add docs and an integration test of the counter and you should be good to go!

/wait

**Review by alyssawilk — APPROVED** [review]
Sweet!

**Comment by alyssawilk** [comment]
I think this needs a main merge to fix CI, otherwise should be good to go!

**Pull request merged**